src/vnodeTransformers/stubComponentsTransformer.ts

```diff
@@ -36,10 +36,14 @@ interface StubOptions {
   renderStubDefaultSlot?: boolean
 }
 
-const doNotStubComponents: WeakSet<ConcreteComponent> = new WeakSet()
+const doNotStubComponents = new WeakMap<ConcreteComponent, boolean>()
 const shouldNotStub = (type: ConcreteComponent) => doNotStubComponents.has(type)
-export const addToDoNotStubComponents = (type: ConcreteComponent) =>
-  doNotStubComponents.add(type)
+const shouldNotStubRoot = (type: ConcreteComponent) =>
+  !!doNotStubComponents.get(type)
+export const addToDoNotStubComponents = (
+  type: ConcreteComponent,
+  onlyRoot?: boolean
+) => doNotStubComponents.set(type, !!onlyRoot)
```

Comment:
maybe we could have two distinct functions, with different names to make it easier to understand?
like `addToDoNotStubComponentsButOnlyInRoot`? (I admit it's not a great name ^^)